encoding/wkt/wkt_encode.go

```diff
@@ -4,65 +4,91 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"reflect"
 	"strconv"
 
 	"github.com/go-spatial/geom"
+	"github.com/go-spatial/geom/cmp"
 )
 
-func isNil(a interface{}) bool {
-	defer func() { recover() }()
-	return a == nil || reflect.ValueOf(a).IsNil()
-}
-
 type Encoder struct {
 	w io.Writer
 	fbuf []byte
+
+	// Strict causes the encoder to return errors if the geometries have empty
+	// sub geometries where not allowed by the wkt spec. When Strict
+	// false, empty geometries are ignored.
+	//		Point: can be empty
+	//		MultiPoint: can have empty points
+	//		LineString: cannot have empty points
+	//		MultiLineString: can have empty line strings non-empty line strings cannot have empty points
+	//		Polygon: cannot have empty line strings, non-empty line strings cannot have empty points
+	//		MultiPolygon: can have empty linestrings, polygons cannot have empty linestrings, line strings cannot have empty points
+	//		Collection: can have empty geometries
+	Strict bool
+
+	// The precision that is passed into strconv.FormatFloat
+	// https://golang.org/pkg/strconv/#FormatFloat
+	Precision *int
+
+	// The format flag that is passed into strconv.FormatFloat. If
+	// performance is an issue then, the 'E' or 'e' flag is recommended.
+	// https://golang.org/pkg/strconv/#FormatFloat
+	Fmt byte
 }
 
+var (
+	DefaultEncoderStrict = true
+	DefaultEncoderPrecision = 5
+	DefaultEncoderFmt = byte('g')
+)
+
 func NewEncoder(w io.Writer) *Encoder {
-	// for reference, the min value of EPSG:3857 -20026376.390 -> 13 characters
+	// our float would be one of
+	// ddd.ddd with Precision+1 number of d's => Precision + 1 + 1
+	// d.ddde+xx with Precision+1 number of d's => Precision + 1 + 5
+	prec := DefaultEncoderPrecision
```

Comment:
I'm thinking expose the Strict, Precision and Fmt here in the `NewEncoder` and make it private in the Encoder. It does not make sense to change it after the fact. Also, this would allow you make Precision a concreate value, instead. 

Make another function `NewDefaultEncoder(w io.Writer)` that would be this function.

Then you can get rid of the DefaultEncoderStrict, DefaultEncoderPrecision, and DefaultEncoderFmt values. 
And instead have:
```go
var (
       defaultEncoder Encoder{
             precision: 5,
             fmt: byte('g'),
       }
)
```

Also, by default, it should not be strict.

Why are we passing around pointers to Encoder? We don't mutate anything other than `io.Writer`, which is already a pointer. I think you should be hanging everything off a concreate type of `Encoder`.  `fbuf` is the same as `io.Writer`.

Reply:
I was passing around pointers as a default, I'll make those changes.